### ISSUE TITLE
Issues setting up a Software Defined Network (SDN) infrastructure in the VMM fabric

### ISSUE THREAD

**Issue by nshetty84** [Microsoft/SDN#74]
We are following the step by step article given below trying to setup the SDN infrastructure in the VMM 2016 fabric on Windows Server 2016 and we are facing a rather weird issue with the same:
https://blogs.technet.microsoft.com/larryexchange/2016/09/05/configure-wap-to-support-new-sdn-stack-on-windows-server-2016/ 
and
https://technet.microsoft.com/en-us/system-center-docs/vmm/scenario/sdn-overview
We are able to setup the network controller vms fine but when the Pre-install scripts run, we get the below given errors:

Error (22042)
The service <servicename> was not successfully deployed. Review the event log to determine the cause and corrective actions.

Recommended Action
The deployment can be restarted by retrying the job.

Error (22753)
The script command with properties: Type (PreInstall), Deployment Order (5) and Parent Type (ApplicationProfile), failed to complete successfully. Refer to the errors list for more information.

Recommended Action
If the script command's job restart action is set to restart, then the script will be re-executed.  Otherwise, the script command will be skipped when the job is restarted, in which case corrective action should be taken to mitigate the effects of the script command failure.

Error (22753)
The script command with properties: Type (PreInstall), Deployment Order (5) and Parent Type (ApplicationProfile), failed to complete successfully. Refer to the errors list for more information.

Recommended Action
If the script command's job restart action is set to restart, then the script will be re-executed.  Otherwise, the script command will be skipped when the job is restarted, in which case corrective action should be taken to mitigate the effects of the script command failure.

Error (22632)
The script command standard error matched the failure policy setting "Match any string" with its result The string is missing the terminator: '.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : TerminatorExpectedAtEndOfString
 
'joT'' is not recognized as an internal or external command,
operable program or batch file.. For more information, see the standard error log C:\NCInstall\InstallNetworkController-AllNodes.err.

From the looks of it, it seems that somewhere in the script a terminator is missing but we are not able to figure out the same. Any help would be appreciated.

**Comment by JMesser81** [comment]
ping @manishmsft 

I've seen this error in the past when multiple network controller certificate (pfx) files are included in the ServerCertificate.cr or NCCertificate.cr directories

**Comment by nshetty84** [comment]
We spent close to 20 hrs working with PSS and the iDNA trace revealed the reason why we faced this issue. Believe it or not, it was because the password for "MgmtDomainAccount" variable was being passed as a clear-text and since it had an ampersand (&) in the same, Powershell used to truncate everything after the same and picked it up as a normal string character and hence expected a terminator. Yes, it is stupid...can't believe that we spent so many hours and the reason could be so unbelievable. We had to use a non-complex password to workaround the issue.
